cpp/unreal_plugins/WheeledVehicle/Source/WheeledVehicle/VehiclePawn.cpp

```diff
@@ -0,0 +1,177 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "WheeledVehicle/VehiclePawn.h"
+
+#include <string>
+
+#include <Eigen/Dense>
+
+#include <Animation/AnimInstance.h>
+#include <Camera/CameraComponent.h>
+#include <Components/BoxComponent.h>
+#include <Components/InputComponent.h>
+#include <Components/SkeletalMeshComponent.h>
+#include <Engine/CollisionProfile.h>
+#include <UObject/ConstructorHelpers.h>
+
+#include "CoreUtils/Assert.h"
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Log.h"
+#include "CoreUtils/Unreal.h"
+#include "WheeledVehicle/VehicleMovementComponent.h"
+
+AVehiclePawn::AVehiclePawn(const FObjectInitializer& object_initializer) : APawn(object_initializer)
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    if (!Config::s_initialized_) {
+        return;
+    }
+
+    ConstructorHelpers::FObjectFinder<USkeletalMesh> skeletal_mesh(*Unreal::toFString(Config::get<std::string>("WHEELED_VEHICLE.VEHICLE_PAWN.SKELETAL_MESH")));
+    SP_ASSERT(skeletal_mesh.Succeeded());
+
+    ConstructorHelpers::FClassFinder<UAnimInstance> anim_instance(*Unreal::toFString(Config::get<std::string>("WHEELED_VEHICLE.VEHICLE_PAWN.ANIM_INSTANCE")));
+    SP_ASSERT(anim_instance.Succeeded());
+
+    skeletal_mesh_component_ = CreateDefaultSubobject<USkeletalMeshComponent>(TEXT("AVehiclePawn::skeletal_mesh_component_"));
+    SP_ASSERT(skeletal_mesh_component_);
+    skeletal_mesh_component_->SetSkeletalMesh(skeletal_mesh.Object);
+    skeletal_mesh_component_->SetAnimClass(anim_instance.Class);
+    skeletal_mesh_component_->SetCollisionProfileName(UCollisionProfile::Vehicle_ProfileName);
+    skeletal_mesh_component_->BodyInstance.bSimulatePhysics = true;
```

Comment:
Which of these parameters are explicitly required by the Chaos vehicle tutorials? For example, I suspect bUseCCD is not required.

Reply:
Chaos tutorial uses `AWheeledVehiclePawn` (https://github.com/EpicGames/UnrealEngine/blob/5.2.0-release/Engine/Plugins/Experimental/ChaosVehiclesPlugin/Source/ChaosVehicles/Private/WheeledVehiclePawn.cpp). The parameters here are added to mimic that implementation.